graphql_compiler/schema_transformation/split_query.py

```diff
@@ -122,11 +123,7 @@ def _split_query_ast_one_level_recursive_type_coercion(
         list of selections in the SelectionSet one level above. If no changes were made, the
         exact input list object will be returned
     """
-    if len(selections) != 1:
-        raise AssertionError(u'Selections {} should only contain one element'.format(selections))
-    type_coercion = selections[0]
-    if not isinstance(type_coercion, InlineFragment):
-        raise AssertionError(u'Selection {} should be an inline fragment.'.format(type_coercion))
+    type_coercion = try_get_inline_fragment(selections)
```

Comment:
Seems like here we implicitly expect `type_coercion` to not be `None`.

But when we call this function, we already run `try_get_inline_fragment(selections)`, though we just check the result for `None`. So how about we do a small refactor and accept a `type_coercion_ast` argument here, and pass the `try_get_inline_fragment(selections)` result from the caller into here. That way we don't have to worry about accidentally getting a bad `selections` object. We could then return only the new (or old) type coercion, and handle the selections-magic in the caller.

Reply:
There's very little code happening in the type coercion case, so I just got rid of the extra function and now do stuff directly in the caller.